design/erasure-coding.md

```diff
@@ -0,0 +1,83 @@
+3) Repeat the above steps for the length of the entire dataset.
+
+Bellow is a graphical outline of the process:
+
+_The sequence of original blocks_
+
+![](./figs/blocks.svg)
+
+_The logical matrix resulting from stacking each $S$ symbols together, where $K=3$ and $S=4$_
+
+![](./figs/matrix1.svg)
+
+_Symbols and Coding direction. Each cell corresponds to a  symbol $s$ and each column of symbols are coded together_
+
+![](./figs/matrix3.svg)
+
+_The resulting matrix with parity blocks added, where $K=3$, $M=2$, $N=5$ and $S=4$_
+
+![](./figs/matrix2.svg)
+
+The resulting structure is a matrix of height $N$ and width $S$. It allows loosing any $M$ "rows" of the matrix and still reconstructing the original dataset. Of course, loosing more than $M$ symbols in a given column would still render the entire dataset invalid, but the likelihood of that happening is mitigated by placing the individual original and parity blocks on independent locations.
+
+Moreover, the resulting dataset is still systematic, which means that the original blocks are unchanged and can be used without prior decoding.
+
+## Data placement and erasures
+
+The code introduced in the above section satisfies our original requirement of any $K$ blocks allowing to reconstruct the original dataset. In fact, one can easily see that every row in the resulting matrix is a standalone element and any $K$ rows will allow reconstructing the original dataset.
+
+However, the code is $K$ rows strong only if we operate under the assumption that each failure is independent of each other. Thus, it is a requirement that each row is placed independently. Moreover, the overall strength of the code decreases based on the number of dependent failures. If we place $N$ rows on two independent locations, we can only tolerate $M=N/2$ failures, three will allow tolerating $M=N/3$ failures and so on. Hence, the code is only as strong as the number of independent locations each element is stored on. Thus, each row and better yet, each element (block) of the matrix should be stored independently and in a pattern mitigating dependence, meaning that placing elements of the same column together should be avoided.
+
+### Load balancing retrieval
+
+There is an issue resulting mostly from the systematic nature of the code and the presence of original and parity data.
+
+It is safe to assume that the systematic (original) data is going to be retrieved more frequently than the parity data, which should only be retrieved when one of the systematic pieces have gone missing thus, in order to prevent congestion and overloading of the systematic nodes some further placement considerations should be taken in to account. Namely, having nodes dedicated to parity data should be avoided.
```

Comment:
one of the systematic pieces has gone missing..

Reply:
Not sure I understand what the proposed change is?